toolkit/tools/internal/safemount/safemount.go

```diff
@@ -47,13 +48,33 @@ func (m *Mount) newMountHelper(source, target, fstype string, flags uintptr, dat
 		source, target, fstype, flags, data)
 
 	if makeAndDeleteDir {
-		// Create the mount target directory.
-		err = os.MkdirAll(target, os.ModePerm)
+		sourceInfo, err := os.Stat(source)
 		if err != nil {
-			return fmt.Errorf("failed to create mount directory (%s):\n%w", target, err)
+			return fmt.Errorf("failed to stat mount source (%s):\n%w", source, err)
 		}
 
-		m.dirCreated = true
+		isFile := (sourceInfo.Mode() & os.ModeType) == 0
+		if !isFile {
+			// Create the mount target directory.
+			err = os.MkdirAll(target, os.ModePerm)
+			if err != nil {
+				return fmt.Errorf("failed to create mount directory (%s):\n%w", target, err)
+			}
+		} else {
+			// Create parent directory.
+			err = os.MkdirAll(filepath.Dir(target), os.ModePerm)
+			if err != nil {
+				return fmt.Errorf("failed to create mount target parent directory (%s):\n%w", target, err)
+			}
+
+			// Create placeholder file.
+			err = os.WriteFile(target, []byte(nil), os.ModePerm)
```

Comment:
`[]byte{}` simplify to create the empty file?

Reply:
In Go, `nil` is considered to be an empty list. So, `[]byte(nil)` and `[]byte{}` are functionally the same thing.